packages/pyright-internal/src/analyzer/analyzerNodeInfo.ts

```diff
@@ -228,6 +228,11 @@ export function isCodeUnreachable(node: ParseNode): boolean {
     return false;
 }
 
+export function hasAnalyzerInfo(node: ParseNode): boolean {
+    const info = getAnalyzerInfo(node);
+    return info?.fileInfo? true : false;
```

Comment:
I think we should jsut check whether `info` is there, not `fileInfo` or change the function name to `hasFileInfo`?

Reply:
The `fileInfo` subfield will appear only in the top-most parse node in a tree, so this isn't the correct check.

In any case, I don't think this is the right way to address this issue. If we're trying to protect against situations where binding failed, we need to do so in a more sound manner rather than just adding a spot check in this one place for some symptom of the underlying condition.

There are many places in the code where we assume that a bound file has a valid fileInfo at the topmost parse node in its tree. If this assumption is violated, it's potentially going to cause other crashes everywhere else it is accessed.

Please do not check in this change.

Reply:
we are open to any other suggestion. another way is making sure that `fileinfo` is always set even when exception case.

Reply:
I think we need to understand the root cause better. Think about ways we can gather more information through telemetry or in the crash data. 

In the exception case, we could verify that `fileInfo` is set and assert (and therefore force a crash) if it's not. That would allow us to validate your theory and potentially get us one step closer to root causing the problem. I don't think we should just mask the problem and move on.